modules/sagemaker/sagemaker-templates-service-catalog/templates/model_deploy/github_repo_construct.py

```diff
@@ -0,0 +1,284 @@
+            'Accept': 'application/vnd.github.v3+json',
+            'Content-Type': 'application/json'
+        }
+        data = json.dumps({
+            'name': repo_name,
+            'description': repo_description,
+            'private': True
+        }).encode('utf-8')
+
+        req = urllib.request.Request('https://api.github.com/user/repos', data=data, headers=headers, method='POST')
+
+        try:
+            with urllib.request.urlopen(req) as response:
+                repo_data = json.loads(response.read().decode())
+                repo_url = repo_data['clone_url']
+        except urllib.error.HTTPError as e:
+            print(f"HTTP Error: {e.code} - {e.reason}")
+            print(e.read().decode())
+            raise
+
+        print(f"Repository created: {repo_url}")
+
+        # Download entire S3 bucket
+        s3 = boto3.client('s3')
+        with tempfile.TemporaryDirectory() as tmp_dir:
+
+            # Generate a unique directory name
+            # random_suffix = generate_random_string()
+            # tmp_dir = f"/tmp/repo_{random_suffix}"
+            # zip_file_path = f"{tmp_dir}/deploy.zip"
+            zip_file_path = os.path.join(tmp_dir, 'code.zip')
+
+            # download_s3_bucket_v2(s3_bucket_name, tmp_dir)
+            download_s3_object_v3(s3_bucket_name, s3_bucket_object_key, zip_file_path)
+            extract_zip_file(zip_file_path, tmp_dir)
+
+            # Update repo_url with github_owner and github_token
+            repo_url = f"https://{github_owner}:{github_token}@github.com/{github_owner}/{repo_name}.git"
+            # Initialize Git repo and push
+            init_and_push_repo(repo_url, tmp_dir)
+
+            print(f"Repository initialized and pushed successfully")
+
+            # Retrieve the list of existing source credentials from codebuild
+            codebuild = boto3.client('codebuild')
+            response = codebuild.list_source_credentials()
+
+            existing_source_credentials = response['sourceCredentialsInfos']
+            # Check if a source credential for GitHub already exists
+            github_source_credential = any(cred['serverType'] == 'GITHUB' for cred in existing_source_credentials)
+            print(f"GitHub source credential already exists: {github_source_credential}")
+
+            if github_source_credential is False:
+                codebuild.import_source_credentials(
+                    token=code_connection_arn,
+                    serverType="GITHUB",
+                    authType="CODECONNECTIONS"
+                )
+                print(f"GitHub source credential imported successfully")
+            else:
+                print(f"GitHub source credential already exists")
+
+            response_data = {
+                'RepoName': repo_name,
+                'RepoUrl': f"https://github.com/{github_owner}/{repo_name}.git",
+                'GitHubSourceCredential': github_source_credential
```

Comment:
Just checking this is not username/password in cleartext? Do we need to send this in the cfn response?

Reply:
`GitHubSourceCredential` is no more required. I just removed.